### PR TITLE
[MMCA-4540] - Update version, configuration & warnings for CDS Financials service - Jan'24 - V5b

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Play 3.0 upgrade including other latest library upgrades.

Below has been covered as part of this PR

- Play upgrade to 3.0.2
- latest library upgrades to the latest as per catalogue
- remove scalastyle and compilation warnings
- add scalastyle config for tests
- update scalastyle config
- add Constants file and test class
- update tests and refactoring
- minor refactoring

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
 $hmrc-assets-path: "/customs/payment-records/assets/lib/hmrc-frontend/hmrc/";
 
-@import "lib/govuk-frontend/govuk/base";
+@import "lib/govuk-frontend/dist/govuk/base";
 
 @import "components/notifications-bar";
 @import "utils/custom";
 @import "components/progress-bar";
 @import "components/pager";
 @import "components/custom-card";
```
Comment (AdamCoops) [inline app/assets/stylesheets/application.scss:10]: Does this file have an empty line at the end?

File: app/controllers/AuthorizedToViewController.scala [hunk 2]
```diff
@@ -274,14 +282,14 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
...
     searchedAuthorities.authorities.exists {
       case AuthorisedDutyDefermentAccount(_, balances) => balances.map(_.periodAvailableAccountBalance).isEmpty
       case AuthorisedCashAccount(_, availableAccountBalance) => availableAccountBalance.isEmpty
       case AuthorisedGeneralGuaranteeAccount(_, availableGuaranteeBalance) => availableGuaranteeBalance.isEmpty
     }
   }
 
-  protected def stripWithWhitespace(str: String): String =
+  private def stripWithWhitespace(str: String): String =
     str.replaceAll("\\s", emptyString).toUpperCase
```
Comment (AdamCoops) [inline app/controllers/AuthorizedToViewController.scala:294]: does this need to be on a separate line? I don't think it has length issues

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (tyagikumarvineet) [reply app/assets/stylesheets/application.scss:10]: Yes, it has.

File: app/controllers/AuthorizedRequestReceivedController.scala [hunk 3]
```diff
@@ -36,8 +36,8 @@ class AuthorizedRequestReceivedController @Inject()(authenticate: IdentifierActi
                                                     customsDataStore: DataStoreService,
                                                     implicit val mcc: MessagesControllerComponents,
                                                     authorisedToViewRequestReceived: authorised_to_view_request_received,
```
Comment (AdamCoops) [inline app/controllers/AuthorizedRequestReceivedController.scala:38]: This line (38) is slightly to long I think

File: app/controllers/AuthorizedToViewController.scala [hunk 4]
```diff
@@ -50,8 +50,8 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
                                            authorisedToViewSearch: authorised_to_view_search,
                                            authorisedToViewSearchResult: authorised_to_view_search_result,
                                            authorisedToViewSearchNoResult: authorised_to_view_search_no_result,
-                                           eoriNumberFormProvider: EoriNumberFormProvider)(
```
Comment (AdamCoops) [inline app/controllers/AuthorizedToViewController.scala:53]: on line 40 of this file we have a scalastyle off comment, is this still required?

File: app/domain/CDSAccount.scala [hunk 5]
```diff
@@ -38,37 +38,34 @@ case class DutyDefermentAccount(number: String,
 
   def displayBalances(periodAccountLimit: Option[BigDecimal],
                       periodGuaranteeLimit: Option[BigDecimal],
-                      periodGuaranteeLimitRemaining: Option[BigDecimal]
```
Comment (AdamCoops) [inline app/domain/CDSAccount.scala:41]: There is a Scalastyle comment in this file (near the top). just checking its required

**Comment by AdamCoops** [comment]
In SdesFile we have a comment for // scalastyle:off number.of.types.
Is this still required or can we refactor it out? might be a future ticket to remove all the scalastyle comments

**Comment by AdamCoops** [comment]
There is another scala comment in Form.Mappings.Constraints

File: app/controllers/AuthorizedToViewController.scala (hunk 2 above)
Comment (tyagikumarvineet) [reply app/controllers/AuthorizedToViewController.scala:294]: Done, made one single line

File: app/controllers/AuthorizedToViewController.scala (hunk 4 above)
Comment (tyagikumarvineet) [reply app/controllers/AuthorizedToViewController.scala:53]: yes, its is required. It needs to be refactored to reduce the cyclomatic complexity. Should be done as part of separate ticket

File: app/controllers/AuthorizedToViewController.scala [hunk 6]
```diff
@@ -100,9 +100,8 @@ class AuthorizedToViewController @Inject()(authenticate: IdentifierAction,
   }
 
   private def processSearchQuery(request: AuthenticatedRequest[AnyContent],
-                                 query: EORI)(
-                                  implicit hc: HeaderCarrier,
-                                  messages: Messages, appConfig: AppConfig): Future[Result] = {
+                                 query: EORI)
+                                (implicit hc: HeaderCarrier, messages: Messages, appConfig: AppConfig): Future[Result] = {
```
[the remark this answers is not in the document]
Comment (tyagikumarvineet) [reply app/controllers/AuthorizedToViewController.scala:104]: Done

File: app/domain/CDSAccount.scala (hunk 5 above)
Comment (tyagikumarvineet) [reply app/domain/CDSAccount.scala:41]: Removed, seems not required now

**Comment by tyagikumarvineet** [comment]
> SdesFile

Yes, it is required. Yes, we should create a separate ticket as it may involve big refactoring as it relates to the declared types exceeding 30

**Comment by tyagikumarvineet** [comment]
> There is another scala comment in Form.Mappings.Constraints

Yes, it is required. Refactoring is required as part of separate ticket

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**